File: cloud_testing/cloud_api_poc/src/main/java/cloud_api_poc/GcToolResults.java
package cloud_api_poc;

import static cloud_api_poc.Config.getCredential;
import static cloud_api_poc.Constants.*;

import com.google.api.services.toolresults.ToolResults;

public abstract class GcToolResults {
  private GcToolResults() {}

  private static ToolResults toolresults = null;

  static {
    toolresults =
        new ToolResults.Builder(httpTransport, JSON_FACTORY, getCredential())
            .setApplicationName(applicationName)
            .build();
  }

  public static ToolResults get() {
    return toolresults;
  }
}

File: cloud_testing/kotlin_poc/src/main/kotlin/GcToolResults.kt
import com.google.api.services.toolresults.ToolResults

import Config.credential
import Constants.httpTransport
import Constants.JSON_FACTORY
import Constants.applicationName

object GcToolResults {

    private var toolresults: ToolResults? = null

    init {
        toolresults = ToolResults.Builder(httpTransport!!, JSON_FACTORY, credential)
                .setApplicationName(applicationName)
                .build()
    }

    fun get(): ToolResults? {
        return toolresults
    }
}
